common/static/common/js/discussion/discussion_router.js

```diff
@@ -102,11 +108,43 @@
             };
 
             DiscussionRouter.prototype.showThread = function(forum_name, thread_id) {
+                var self, callback;
+                self = this;
                 this.thread = this.discussion.get(thread_id);
+
+                if (!this.thread){
```

Comment:
@bradenmacdonald Unfortunately that's one feature @Kelketek implemented, and there aren't any references to PR on the [commit](https://github.com/edx-solutions/edx-platform/commit/e28dde32169b388235030dba945366f4c13e565f) for some reason. So the motivation is unknown (yet).

@Kelketek is there any chance you still remember what was the reason for this?